src/js/misc/utility.js

```diff
@@ -412,6 +412,33 @@ function merge_with_defaults(obj) {
 
 MG.merge_with_defaults = merge_with_defaults;
 
+function options_to_defaults(obj) {
+  return Object.keys(obj).reduce((r, k) => {
+    r[k] = obj[k][0];
+    return r;
+  }, {});
+}
+
+function compare_type(type, value) {
+  if (value == null) return true; // allow null or undefined
+  if (typeof type === 'string') {
+    if (type.substr(-2) === '[]') {
+      if (!is_array(value)) return false;
+      return value.every(i => compare_type(type.slice(0, -2), i));
+    }
+    if (typeof value === type || value === type) return true;
+    return false;
+  }
+  return is_array(type) && !!~type.findIndex(i => compare_type(i, value, true));
+}
+
+function mg_validate_option(key, value) {
+  if (!is_array(MG.options[key])) return false; // not existed option
+  const typeDef = MG.options[key][1];
+  if (!typeDef) return true; // not restricted type
+  return compare_type(typeDef, value);
+}
```

Comment:
I think this method is great. Do you think you could call it automatically when the `data_graphic` method is called, and emit a `console.warning` if a parameter does not match the expected type or is not recognized?